nvflare/fuel/hci/client/cli.py

```diff
@@ -206,6 +220,14 @@ def complete(self, text, state):
         results = [x + " " for x in self.api.all_cmds if x.startswith(text)] + [None]
         return results[state]
 
+    def preloop(self):
+        if readline and self.cli_history_file and os.path.exists(self.cli_history_file):
+            readline.read_history_file(self.cli_history_file)
+
+    def postloop(self):
```

Comment:
So if the client crashed before exiting the loop, history won't be written?

Reply:
Ah yes, precmd() and postcmd() would probably be better then.